Varlink/nmstate_varlink.py

```diff
@@ -0,0 +1,242 @@
+        """
+        if state is None:
+            raise varlink.InvalidParameter("state")
+        apply_kwargs = {}
+        if verify_change is not None:
+            apply_kwargs["verify_change"] = verify_change
+        if commit is not None:
+            apply_kwargs["commit"] = commit
+        if rollback_timeout is not None:
+            apply_kwargs["rollback_timeout"] = rollback_timeout
+        try:
+            libnmstate.apply(state, **apply_kwargs)
+        except TypeError as exception:
+            raise varlink.InvalidParameter
+        except libnmstate.error.NmstatePermissionError as exception:
+            raise NmstatePermissionError(str(exception))
+        except libnmstate.error.NmstateConflictError as exception:
+            raise NmstateConflictError(str(exception))
+        except varlink.InvalidParameter as exception:
+            raise varlink.InvalidParameter
+
+    def Commit(self, checkpoint):
+        """
+        Commits the checkpoint
+        """
+        commit_kwargs = {}
+        if checkpoint is not None:
+            commit_kwargs["checkpoint"] = checkpoint
+        try:
+            libnmstate.commit(**commit_kwargs)
+        except libnmstate.error.NmstateValueError as exception:
+            raise NmstateValueError(str(exception))
+
+    def Rollback(self, checkpoint):
+        """
+        Roll back to the checkpoint
+        """
+        rollback_kwargs = {}
+        if checkpoint is not None:
+            rollback_kwargs["checkpoint"] = checkpoint
+        try:
+            libnmstate.rollback(**rollback_kwargs)
+        except libnmstate.error.NmstateValueError as exception:
+            raise NmstateValueError(str(exception))
+
+
+def get_varlink_server(address):
+    """
+    Returns varlink server object
+    Checks if the varlink address already in use.
+    """
+    server = varlink.ThreadingServer(
+        address, ServiceRequestHandler, bind_and_activate=False
+    )
+    server.allow_reuse_address = True
+    try:
+        server.server_bind()
+        server.server_activate()
+    except OSError:
+        server.server_close()
+        server = varlink.ThreadingServer(
+            address, ServiceRequestHandler, bind_and_activate=True
+        )
+        server.allow_reuse_address = True
+    return server
+
+
+def run_server(address):
+    """
+    Runs the varlink server in the specified the address
+    uses unix socket address and tcp address
+    """
+    varlink_server = get_varlink_server(address)
+    try:
+        with varlink_server as server:
+            logging.warning(
+                "Listening address : %s", str(server.server_address)
+            )
+            server.serve_forever()
+    except KeyboardInterrupt:
+        varlink_server.shutdown()
+    finally:
+        varlink_server.shutdown()
+        varlink_server.server_close()
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description="Varlink io.nmstate")
```

Comment:
Please move this to a main() method and just accept the address without any flags, so

`nmstate-varlink unix:@nmstate`
`nmstate-varlink tcp:127.0.0.1:1337`
work. Also, the address should be mandatory for now. In the future it should use stdin/stdout in my opinion.